test/framework/deployment_helpers.go

```diff
@@ -95,34 +95,69 @@ func DescribeFailedDeployment(input WaitForDeploymentsAvailableInput, deployment
 	return b.String()
 }
 
-// WatchDeploymentLogsInput is the input for WatchDeploymentLogs.
-type WatchDeploymentLogsInput struct {
+// logMetadata contains metadata about the logs.
+// The format is very similar to the one used by promtail.
+type logMetadata struct {
```

Comment:
Can you please also move logMetadata below watchPodLogs. It's a bit out of place here. Below the func because you could see it as some sort of util used by the func (and not above because it's not an input parameter)

Reply:
@adilGhaffarDev last one from my side, otherwise lgtm